pandas/tests/scalar/timestamp/test_timestamp.py

```diff
@@ -700,6 +701,18 @@ class SubDatetime(datetime):
         expected = Timestamp(2000, 1, 1)
         assert result == expected
 
+    @pytest.mark.skipif(
+        not compat.PY38,
+        reason="datetime.fromisocalendar was added in Python version 3.8",
+    )
+    def test_constructor_fromisocalendar(self):
+        # GH 30395
+        expected_timestamp = Timestamp("2000-01-03 00:00:00")
+        expected_stdlib = datetime.fromisocalendar(2000, 1, 1)
+        result = Timestamp.fromisocalendar(2000, 1, 1)
+        assert result == expected_timestamp
+        assert result == expected_stdlib
```

Comment:
one more thing i forgot, this needs to `assert isinstance(result, Timestamp)`.  Otherwise i think this test would pass in master